### DISCUSSION TITLE
How to change the default port 8000

### DISCUSSION THREAD

**Discussion by 0PandaDEV** [supabase/supabase#20683] [Questions]
How can I change the default port for Kong from 8000 to for example 8001 because I have another service already using 8000. I could not find any information on this in the docs or somewhere on the internet.

For reference, I use Docker to self-host a Supabase instance on an Ubuntu Server 23.10.

I tried changing the ports in the docker compose file, but that didn't help at all.

**Comment by anandkkpr** [comment]
This is also very important if you are trying to develop TWO Supabase Projects simultaneously on a single development machine. In _therory_, we are able to star up multiple Supabase instances BUT, in our case, since one instance is where we host our DB and the other instance is where we host our Edge Functions (for sec purposes), we are unable to do this successfully.

We had thought we could simply make use of `Deno.env.get('SUPABASE_URL')` within our Edge Functions Supabase instance to point to the Database Supabase instance - e.g.: `http://127.0.0.1:54421` (where our Edge Functions Instance's API URL is the default `http://127.0.0.1:54321`) but **this did not work**!

After a LOT of digging, poking and prodding, I came across the same issue from a different angle. I found that IF we want to change the port that Docker is using for the Kong API Gateway, we would actually need to **drop the convenience of the Supabase CLI** and actually **manually** setup Supabase instances directly using the documentation here: [Self-Hosting with Docker](https://supabase.com/docs/guides/self-hosting/docker). Using this method, we can then edit the `docker-compose.yml` file so that each instance uses a different port **that we control** from this file.

The thing is, that is way-overkill for simply wanting to test and develop Edge Functions that run DB queries against a **second** Supabase instance (setup auto-magically by `supabase start` and linked to its PRD project via `supabase link --project-ref ABCD`).

Some help from Supabase - maybe adding "Docker Kong Port" a config option in `supabase/config.toml` would be REALLY helpful and appreciated!!! I think it would greatly improve flexibility for local development, esp. for orgs that have have chosen to host various Supabase services (e.g. Edge Functions, Database, Auth, S3, etc) in separate projects.

I hope all this info is clear and easily understandable! Please let me know if more clarity is required/desired!

**Comment by Jack74r** [comment]
Did you solve your issue ?

**Reply by anandkkpr** [reply]
@Jack74r - not as originally hoped.

We are migrating edge functions to be in the same Project/instance as the database. In our case, it isn't worth the effort to try to keep them separate.

We're still going that Supabase will update the CLI toolchain to enable easy access configuration via `config.toml` that allows users/devs to specify the port config that is passed to Kong by Docker.

**Comment by youcefmegoura** [comment]
Hello, using v1.24.09, docker-compose.yml file : 
https://github.com/supabase/supabase/blob/v1.24.09/docker/docker-compose.yml

``` yaml
[...]
services:
[...]
  kong:
    container_name: supabase-kong
    image: kong:2.8.1
    restart: unless-stopped
    ports:
      - ${KONG_HTTP_PORT}:8000/tcp     <== change this to some thing like this: 8001:8000/tcp
      - ${KONG_HTTPS_PORT}:8443/tcp 
[...]
```

**Reply by digiSal** [reply]
i had to have mine setup like this
change {KONG_HTTP_PORT}:8001:8000/tcp 
to - "8001:8000/tcp"

to use port 8001. everything else stayed the same.

**Comment by aantti** [comment]
While not exactly the same discussion in #40686 - still leaving it here for visibility. There was a lot of details discussed in that issue regarding Kong configuration, ports different from the default 8000, etc.
